modules/auxiliary/scanner/http/wordpress_scanner.rb

```diff
@@ -10,28 +10,83 @@ class MetasploitModule < Msf::Auxiliary
+        'h00die' # plugins and themes
+      ],
+      'License' => MSF_LICENSE
     )
+    register_options [
+      OptBool.new('THEMES', [false, 'Detect themes', true]),
+      OptBool.new('PLUGINS', [false, 'Detect plugins', true]),
+      OptPath.new('THEMES_FILE', [
+        true, 'File containing themes to enumerate',
+        File.join(Msf::Config.data_directory, 'wordlists', 'wp-themes.txt')
+      ]),
+      OptPath.new('PLUGINS_FILE', [
+        true, 'File containing plugins to enumerate',
+        File.join(Msf::Config.data_directory, 'wordlists', 'wp-plugins.txt')
+      ]),
+      OptInt.new('PROGRESS', [true, 'how often to print progress', 1000])
+    ]
+  end
+
+  def print_progress(i, total)
+    print_status("Progress #{i}/#{total} (#{((i.to_f / total) * 100).truncate(2)}%)")
   end
 
   def run_host(target_host)
     print_status("Trying #{target_host}")
     if wordpress_and_online?
       version = wordpress_version
-      version_string = version ? version : '(no version detected)'
+      version_string = version || '(no version detected)'
       print_good("#{target_host} running Wordpress #{version_string}")
       report_note(
-          {
-              :host   => target_host,
-              :proto  => 'tcp',
-              :sname => (ssl ? 'https' : 'http'),
-              :port   => rport,
-              :type   => "Wordpress #{version_string}",
-              :data   => target_uri
-          })
+        {
+          host: target_host,
+          proto: 'tcp',
+          sname: (ssl ? 'https' : 'http'),
+          port: rport,
+          type: "Wordpress #{version_string}",
+          data: target_uri
+        }
+      )
+      if datastore['THEMES']
+        print_status('Enumerating Themes')
+
+        f = File.open(datastore['THEMES_FILE'], 'rb')
+        total = f.lines.count
+        f.rewind
+        f = f.lines
+        f.each_with_index do |theme, i|
+          theme = theme.strip
+          print_progress(i, total) if i % datastore['PROGRESS'] == 0
+          vprint_status("Checking theme: #{theme}")
+          version = check_theme_version_from_readme(theme)
+          next if version == Msf::Exploit::CheckCode::Unknown # aka not found
+
+          print_good("Detected Theme: #{theme} version #{version.details} ")
+        end
+      end
+      if datastore['PLUGINS']
+        print_status('Enumerating Plugins')
+
+        f = File.open(datastore['PLUGINS_FILE'], 'rb')
+        total = f.lines.count
+        f.rewind
+        f = f.lines
+        f.each_with_index do |plugin, i|
+          plugin = plugin.strip
+          print_progress(i, total) if i % datastore['PROGRESS'] == 0
+          vprint_status("Checking plugin: #{plugin}")
+          version = check_plugin_version_from_readme(plugin)
+          next if version == Msf::Exploit::CheckCode::Unknown # aka not found
+
+          print_good("Detected Plugin: #{plugin} version #{version.details} ")
+        end
```

Comment:
These should both print 100% complete at the end. Right now in your example it says 100% at the end but it's actually for the host. If you're scanning multiple hosts it'd also be nice to include info in the progress output like the target and what is being scanned. That way when the user is scanning multiple systems and the output is mixed, they know what is at what progress.